tests/chainer_tests/functions_tests/rnn_tests/test_function_slstm.py

```diff
@@ -169,6 +170,87 @@ def test_double_backward(self,  backend_config):
         self.check_double_backward(
             self.inputs, self.grad_outputs, self.grad_grad_inputs,
             backend_config)
+"""
+
+
+@testing.parameterize(*testing.product_dict(
+    [
+        {'batch_size': 10, 'n_units': 4},
+        {'batch_size': 8, 'n_units': 5},
+        {'batch_size': 6, 'n_units': 24},
+        {'batch_size': 4, 'n_units': 15},
+    ], [
+        {'dtype': numpy.float16},
+        {'dtype': numpy.float32},
+        {'dtype': numpy.float64},
+    ]
+))
+@testing.fix_random()
+@backend.inject_backend_tests(
+    None,
+    # ChainerX tests
+    testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+    # CPU tests
+    + testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product([
+        [{'use_cuda': True}],
+
+        # Without cuDNN
+        testing.product({
+            'use_cudnn': ['never'],
+        })
+        # With cuDNN
+        + testing.product({
+            'use_cudnn': ['always'],
+            'cudnn_deterministic': [True, False],
+            'autotune': [True, False],
+        })]))
+class TestSLSTM(testing.FunctionTestCase):
```

Comment:
Could you somehow also keep the previous "flat" tests, e.g. using parameterization?